src/pybind/mgr/cephadm/module.py

```diff
@@ -940,15 +943,53 @@ def blink(host, dev, path):
 
         return blink(locs)
 
-    @async_completion
-    def _create_osd(self, all_hosts_, drive_group):
-        all_hosts = orchestrator.InventoryNode.get_host_names(all_hosts_)
-        assert len(drive_group.hosts(all_hosts)) == 1
-        assert len(drive_group.data_devices.paths) > 0
-        assert all(map(lambda p: isinstance(p, six.string_types),
-            drive_group.data_devices.paths))
-
-        host = drive_group.hosts(all_hosts)[0]
+    def call_inventory(self, hosts, drive_groups):
+        def call_create(inventory):
+            return self._prepare_deployment(hosts, drive_groups, inventory)
+
+        return self.get_inventory().then(call_create)
+
+    def create_osds(self, drive_groups):
```

Comment:
`create_osds` is missing a parameter here.

Reply:
I don't think so. `create_osds` shrunk to a single parameter across the codebase (drive_groups)

Reply:
right! remove_osds needs that parameter. I'm sorry.